### PR TITLE
[DOCS] Make audit event types more readable

### PR DESCRIPTION
**Pull request by lockewritesdocs**
The current docs for [audit event types](https://www.elastic.co/guide/en/elasticsearch/reference/7.12/audit-event-types.html) are currently in a tabular format, which makes the descriptions difficult to read and the log examples challenging to parse. This change:

* Puts the event types into a description list for better readability
* Puts examples into a collapsible section, which is easier to read and parse
* Adds `system_access_granted` as an event type
* Includes cross links within the page and to external resources

Preview link: https://elasticsearch_73352.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/audit-event-types.html

### CODE REVIEW

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 1]
```diff
@@ -6,132 +6,390 @@ When you are <<enable-audit-logging,auditing security events>>, a single client
...
+"user.realm":"default_native", "user.roles":["test_role"], "origin.type":
+"rest", "origin.address":"[::1]:52434", "request.id":"yKOgWn2CRQCKYgZRz3phJw",
+"action":"indices:admin/auto_create", "request.name":"CreateIndexRequest", 
+"indices":["<index-{now/d+1d}>"]}
+====
+
+[[event-access-granted]]
+`access_granted`::
+Logged when an authenticated user attempts to execute an action they have the
 necessary privilege to perform. When the `system_access_granted` event is
```
Comment (bytebilly) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:30]: Should we list `system_access_granted` as any other event type in this list? We can still cross-reference it in this description, but I think it's clearer if we provide the full list as expected. I personally got confused by the sentence in the first place.

Comment (lockewritesdocs) [reply x-pack/docs/en/security/auditing/event-types.asciidoc]: Thanks @bytebilly. I added `system_access_granted` as an event type in a54d6bfcaec9f8ffa14ce907355a2fccf5e95b00 based on suggestions from @albertzaharovits.

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Review by albertzaharovits — APPROVED** [review]
LGTM
This is a very nice improvement to our audit docs! Thank you Adam!

**Review by bytebilly — APPROVED** [review]

**Pull request merged**